src/main/kotlin/com/openlattice/shuttle/api/ShuttleApi.kt

```diff
@@ -0,0 +1,97 @@
+package com.openlattice.shuttle.api
+
+import com.openlattice.shuttle.control.Integration
+import com.openlattice.shuttle.control.IntegrationJob
+import com.openlattice.shuttle.control.IntegrationStatus
+import com.openlattice.shuttle.control.IntegrationUpdate
+import retrofit2.http.*
+import java.util.*
+
+
+const val SERVICE = "/shuttle"
+const val CONTROLLER = "/integration"
+const val BASE = SERVICE + CONTROLLER
+
+const val DEFINITION_PATH = "/definition"
+const val STATUS_PATH = "/status"
+
+const val INTEGRATION_NAME = "integrationName"
+const val INTEGRATION_NAME_PATH = "/{$INTEGRATION_NAME}"
+const val JOB_ID = "jobId"
+const val JOB_ID_PATH = "/{$JOB_ID}"
+
+interface ShuttleApi {
+
+    /**
+     * Enqueues an integration on Shuttle Server for a given integration
+     * @param integrationName the name of the integration to be run
+     * @return the unique id of the integration job
+     */
+    @GET(BASE + INTEGRATION_NAME_PATH)
+    fun enqueueIntegration(
+            @Path(INTEGRATION_NAME) integrationName: String
+    ): UUID
+
+    /**
+     * Polls the status of an integration
+     * @param jobId the unique id of the integration job
+     * @return the status of the integration
+     * Note, upon retrieving a final status (i.e. succeeded or failed),
+     * the id/status will be removed from memory
```

Comment:
I feel like removing the status from memory should be a separate operation in case there are multiple people/jobs watching for the status of an integration, and especially if there's an option to poll all integrations